crates/nu-parser/src/parser.rs

```diff
@@ -6174,10 +6174,12 @@ pub fn parse_record(working_set: &mut StateWorkingSet, span: Span) -> Expression
         span_offset: start,
     };
     while !lex_state.input.is_empty() {
-        if lex_state.input[0] == b'}' {
-            extra_tokens = true;
-            unclosed = false;
-            break;
+        if let Some(ParseError::Unbalanced(left, right, _)) = lex_state.error.as_ref() {
+            if left == "{" && right == "}" {
```

Comment:
I do see another place in the parser that checks both `left` and `right` before determining if the error is for unbalanced curly braces, but there's currently no situation where `left` would be `"{"` but `right` would be `"]"` or something. Feels kinda weird to use two separate strings rather than an enum or something, but I guess it doesn't matter that much, and it's out of scope for this PR anyway.